### ISSUE TITLE
translator.clean_cache() errors because services weren't instantiated

### ISSUE THREAD

**Issue by PrajwalVandana** [Animenosekai/translate#104]
Not sure if this package is being actively maintained, but I found an error where
```py
translator = Translator()
translator.clean_cache()
```
errors with
```
TypeError: BaseTranslator.clean_cache() missing 1 required positional argument: 'self'
```
because `translator`'s underlying services have not been instantiated. My workaround was
```py
for i, service in enumerate(translator.services):
    translator._instantiate_translator(service, translator.services, i)
for service in translator.services:
    service.clean_cache()
```
but this still fails on Microsoft and Bing translators (although this part may be my fault, maybe it needs some sort of request argument that I didn't pass; not sure), so I just removed them from my services list.

**Comment by Animenosekai** [comment]
> Not sure if this package is being actively maintained

Yup it is, not super actively because I have a million things on my backlog now lmaooo

<img width="1776" alt="Screenshot 2024-08-04 at 12 48 53" src="https://github.com/user-attachments/assets/a6179c38-f44a-4d90-94cc-73e84dd2827f">

> (couldn't fit everything in here but `translatepy` is the highest in my priority)

> because translator's underlying services have not been instantiated

What version are you on ?

The development version ([v3](https://github.com/Animenosekai/translate/tree/next), on the next branch) does not have this issue :

<img width="1125" alt="Screenshot 2024-08-04 at 12 55 44" src="https://github.com/user-attachments/assets/b0291087-8471-4c52-80e7-93ed6e99144e">

**Comment by PrajwalVandana** [comment]
> not super actively because I have a million things on my backlog now lmaooo

Ahhh, got it, totally fair

I'm on 2.3 (latest release I think); is v3 installable with pip?

**Comment by Animenosekai** [comment]
> I'm on 2.3 (latest release I think); is v3 installable with pip?

Yup, you can install it like so :

```sh
pip install --upgrade git+https://github.com/Animenosekai/translate.git@next
```

**Comment by PrajwalVandana** [comment]
ah cool, would it be safe to make that a dependency for my app?

**Comment by Animenosekai** [comment]
If you want to publish your app, nope it's **not** since it's in an *alpha* state and the API might change a bit.

Also, it hasn't been well tested yet so consider it as a developer beta.

**Comment by PrajwalVandana** [comment]
got it, i'll stick with 2.3 for now then!
